packages/framework-core/src/services/pub-sub/read-model-pub-sub.ts

```diff
@@ -28,41 +29,43 @@ function filterReadModel(readModel: Record<string, any>, filters?: Record<string
   }
   for (const filteredProp in filters) {
     const readModelPropValue = readModel[filteredProp]
-    const { operation, values }: ReadModelPropertyFilter = filters[filteredProp]
+    return filterByOperation(filters[filteredProp], readModelPropValue)
+  }
+  return true
+}
 
+function filterByOperation(filter: ReadModelPropertyFilter, readModelPropValue: any): boolean {
+  for (const [operation, value] of Object.entries(filter as Operation<any>)) {
     switch (operation) {
       case '=':
-        if (readModelPropValue !== values[0]) return false
+        if (readModelPropValue !== value) return false
         break
       case '!=':
-        if (readModelPropValue === values[0]) return false
+        if (readModelPropValue === value) return false
         break
       case '<':
-        if (readModelPropValue >= values[0]) return false
+        if (readModelPropValue >= value) return false
         break
       case '>':
-        if (readModelPropValue <= values[0]) return false
+        if (readModelPropValue <= value) return false
         break
       case '>=':
-        if (readModelPropValue < values[0]) return false
+        if (readModelPropValue < value) return false
         break
       case '<=':
-        if (readModelPropValue > values[0]) return false
+        if (readModelPropValue > value) return false
         break
       case 'in':
-        if (!values.includes(readModelPropValue)) return false
-        break
-      case 'between':
-        if (readModelPropValue < values[0] || readModelPropValue > values[1]) return false
+        if (!value.includes(readModelPropValue)) return false
         break
       case 'contains':
-        if (!contains(readModelPropValue, values[0])) return false
+        if (!contains(readModelPropValue, value)) return false
         break
       case 'not-contains':
-        if (contains(readModelPropValue, values[0])) return false
+        if (contains(readModelPropValue, value)) return false
         break
       case 'begins-with':
-        if (!beginWith(readModelPropValue, values[0] as string)) return false
+        if (!beginWith(readModelPropValue, value as string)) return false
         break
     }
   }
```

Comment:
After the previous changes you can also remove the `return true` at the end of the method.

Reply:
Take into account the previous comment marked with ⚠️ . The read model needs to pass _all the filters_